clients/drcachesim/reader/zipfile_file_reader.cpp

```diff
@@ -204,21 +205,30 @@ file_reader_t<zipfile_reader_t>::skip_thread_instructions(size_t thread_index,
             continue;
         const memref_t &memref = **this;
         if (memref.marker.type == TRACE_TYPE_MARKER) {
-            if (memref.marker.marker_type == TRACE_MARKER_TYPE_TIMESTAMP)
+            if (memref.marker.marker_type == TRACE_MARKER_TYPE_RECORD_ORDINAL) {
+                cur_ref_count_ = memref.marker.marker_value;
+                at_chunk_header = true;
+            } else if (memref.marker.marker_type == TRACE_MARKER_TYPE_TIMESTAMP) {
                 timestamp = entry_copy_;
-            else if (memref.marker.marker_type == TRACE_MARKER_TYPE_CPU_ID)
+                if (at_chunk_header)
+                    cur_ref_count_--; // Invisible.
```

Comment:
So we decrement here because these entries were duplicated from before and aren't really there (invisible), then down below, we have to do it again using suppress_ref_count_ because queues_ would present those entries yet again in read_next_entry. A comment describing why we need both (the decrement and suppress_ref_count_) would help.

Reply:
This would not be needed if this code looked at these at the trace_entry_t level prior to calling process_input_entry -- which it would have to do for the ordinal marker if that is hidden.  Maybe all 3 should be at the trace_entry_t level.  OK I am trying that.